### PR TITLE
fix plot

### LINKED ISSUE #780: trainer plot order
### Environment

deepinv main

### Describe the bug

The behaviour of the deepinv plot function was changed in #684 with now rows and columns of the grid being swapped.

This is what we currently observe when training:
Image

Instead of 
Image

I suspect this being due to the unclear naming convention of torchvision wrt the `n_rows` vairable in the [make_grid](https://docs.pytorch.org/vision/main/generated/torchvision.utils.make_grid.html) function with final shape (batch_size/n_rows, n_rows)...

I'll propose a fix shortly.

### Steps to reproduce

```python3
import deepinv as dinv
a = torch.randn((32, 3, 10, 10))
from deepinv.utils.plotting import prepare_images
out = prepare_images(x=a, y=a, x_net=a, rescale_mode="min_max")
dinv.utils.plot([out[2].unsqueeze(0)])
print(out[2].shape)
> torch.Size([3, 386, 38])
```

### Expected behavior

```python3
print(out[2].shape)
> torch.Size([3, 38, 386])
```

### Full traceback (if applicable)

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by matthieutrs**
Fixes #780 and adds a test to check that plots have correct shape

### Checks to be done before submitting your PR

- [x] `python3 -m pytest deepinv/tests` runs successfully.
- [x] `black .` runs successfully.
- [ ] `make html` runs successfully (in the `docs/` directory).
- [ ] Updated docstrings related to the changes (as applicable).
- [ ] Added an entry to the [CHANGELOG.rst](../CHANGELOG.rst).

### CODE REVIEW

File: deepinv/tests/test_utils.py [hunk 1]
```diff
@@ -900,6 +900,30 @@ def test_prepare_images(x, y, x_net, x_nl, rescale_mode):
...
             isinstance(title, str) for title in titles
         ), "All titles should be strings."
+        assert len(imgs) == len(
+            titles
+        ), "Number of images should match number of titles."
+        assert len(imgs) == sum(v is not None for v in [x, y, x_net, x_nl])
+
+
+@pytest.mark.parametrize(
+    "x",
```
Comment (jscanvic) [inline deepinv/tests/test_utils.py]: Can you instead parametrize by seed (e.g., 0, 1, 2), and set `x = torch.randn(32, 1, 10, 10, generator=rng)` with `rng = torch.Generator().manual_seed(seed)`?

It makes the test deterministic & makes intent clearer imo!

File: deepinv/tests/test_utils.py [hunk 2]
```diff
@@ -900,6 +900,30 @@ def test_prepare_images(x, y, x_net, x_nl, rescale_mode):
         assert all(
             isinstance(title, str) for title in titles
         ), "All titles should be strings."
+        assert len(imgs) == len(
+            titles
+        ), "Number of images should match number of titles."
+        assert len(imgs) == sum(v is not None for v in [x, y, x_net, x_nl])
```
Comment (jscanvic) [inline deepinv/tests/test_utils.py]: It fails because the test to include `y` is not just `y is not None`, but: `y is not None and x is not None and y.shape == x.shape`.

An ugly but working fix (feel free to improve it!):

```suggestion
        assert len(imgs) == sum(v is not None for v in [x, x_net, x_nl]) + int(
            y is not None and x is not None and y.shape == x.shape
        )
```

**Review by jscanvic — CHANGES_REQUESTED** [review]
Really neat fix, thanks Matthieu! Minor comments about the tests & I'll be happy to approve

File: deepinv/tests/test_utils.py (hunk 2 above)
Comment (jscanvic) [reply deepinv/tests/test_utils.py]: Also maybe add an assert message if you can think of a good one! Happy to leave it that way if not

**Review by jscanvic — APPROVED** [review]
LGTM! Thanks Matthieu
